### PR TITLE
Run PHPStan and fix issues

### PR DESCRIPTION
**Pull request by pamtbaau**
I've used the PHPstan config from Grav and ran on level 8. 

Fixed type annotations and bugs, hardened some code and performed some cleanup on the following code:
- classes/Data/*
- classes/RequestHandler.php
- pushy.php (fixed my additions only)

### CODE REVIEW

File: classes/Data/ChangedItems.php [hunk 1]
```diff
@@ -10,15 +10,21 @@
 
 /**
  * Array of urls and their accompanying changedItems
+ * 
+ * @implements \ArrayAccess<string, ChangedItem>
+ * @implements \Iterator<string, ChangedItem>
  */
 class ChangedItems implements ArrayAccess, Countable, Iterator, JsonSerializable
 {
-    /** @var array<string, ChangedItems> */
```
Comment (hughbris) [inline classes/Data/ChangedItems.php:16]: That's funny, I noticed this a few days ago but forgot to raise it with you.

File: classes/Data/ChangedItems.php [hunk 2]
```diff
@@ -27,7 +33,7 @@ public function toArray(): array
             $changedItems[$url] = (array) $changedItem;
         }
 
-        return $changedItem;
```
Comment (hughbris) [inline classes/Data/ChangedItems.php:30]: .. or maybe it was this one

Comment (pamtbaau) [reply classes/Data/ChangedItems.php:30]: Or maybe both... 

Some say you shouldn't use plural forms of variables by adding an `s`. Too prone to mistakes...

Although strong typed languages would have prevented the mistake.

**Comment by pamtbaau** [comment]
Fixed last few phpstan level 9 issues and cleaned up unused code from copy/paste.

**Review by hughbris — APPROVED** [review]
Looks good

**Pull request merged**